test/Microsoft.ML.Functional.Tests/ONNX.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Microsoft.ML.Functional.Tests.Datasets;
+using Microsoft.ML.RunTests;
+using Microsoft.ML.TestFramework;
+using Microsoft.ML.TestFramework.Attributes;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Trainers.FastTree;
+using Microsoft.ML.Transforms;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Functional.Tests
+{
+    public class ONNX : BaseTestClass
+    {
+        public ONNX(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        /// <summary>
+        /// ONNX: I can save a model to ONNX and reload it and use it in a pipeline.
+        /// </summary>
+        [OnnxFactAttribute]
+        public void SaveOnnxModelLoadAndScoreFastTree()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            // Get the dataset.
+            var data = mlContext.Data.LoadFromTextFile<HousingRegression>(GetDataPath(TestDatasets.housing.trainFilename), hasHeader: true);
+
+            // Create a pipeline to train on the housing data.
+            var pipeline = mlContext.Transforms.Concatenate("Features", HousingRegression.Features)
+                .Append(mlContext.Transforms.Normalize("Features"))
+                .AppendCacheCheckpoint(mlContext)
+                .Append(mlContext.Regression.Trainers.FastTree(
+                    new FastTreeRegressionTrainer.Options { NumberOfThreads = 1, NumberOfTrees = 10 }));
```

Comment:
>= 1 [](start = 76, length = 3)

why threads equal to 1? is there a know issue with multi-threading?

Reply:
There are no issues with multithreading. This is for convenience. Namely, we run these sorts of tests single-threaded because
- Many are running at the same time;
- Some algorithms are non-deterministic when multitheaded (e.g. SDCA) and make testing harder (boxing vs exact).

---
In reply to: [267028362](https://github.com/dotnet/machinelearning/pull/2984#discussion_r267028362) [](ancestors = 267028362)